### PR TITLE
Feature/pod priority

### PR DESCRIPTION
**Pull request by maierlars**
Forwad pod priority class to pods.

### CODE REVIEW

File: manifests/templates/test/rbac.yaml [hunk 1]
```diff
@@ -12,8 +12,8 @@ rules:
 - apiGroups: [""]
   resources: ["pods", "services", "persistentvolumes", "persistentvolumeclaims", "secrets", "serviceaccounts", "pods/log"]
   verbs: ["*"]
-- apiGroups: ["apps"]
-  resources: ["daemonsets", "deployments"]
+- apiGroups: ["apps", "scheduling.k8s.io"]
```
Comment (ewoutp) [inline manifests/templates/test/rbac.yaml]: Separate apigroups in different rules. 1 api group per rule

File: docs/Manual/Deployment/Kubernetes/DeploymentResource.md [hunk 2]
```diff
@@ -413,6 +413,10 @@ for each server of this group.
 This setting is not available for group `coordinators`, `syncmasters` & `syncworkers`
 because servers in these groups do not need persistent storage.
 
+### `spec.<group>.priorityClassName: string`
+
+Priority class name for pods of this group. Will be forwarded to the pod spec.
```
Comment (joerg84) [inline docs/Manual/Deployment/Kubernetes/DeploymentResource.md]: Should we add a note that the class should be created upfront (potentially with link to k8s docu)?

File: pkg/util/k8sutil/pods.go [hunk 3]
```diff
@@ -594,6 +599,9 @@ func CreateArangoSyncPod(kubecli kubernetes.Interface, developmentMode bool, dep
 	}
 	p.Spec.Containers = append(p.Spec.Containers, c)
 
+	// Add priorityClassName
+	p.Spec.PriorityClassName = podPriorityClassName
```
Comment (joerg84) [inline pkg/util/k8sutil/pods.go]: Just curious: what is the behavior if you specify a class name of a class which doesn't exist? Do we need to handle something or emit a warning?

Comment (maierlars) [reply pkg/util/k8sutil/pods.go]: Pod creation will fail and a event is attached to the arangodeployment resource containing a proper error message from kubernetes.

**Pull request merged**